index.bs

```diff
@@ -795,6 +743,562 @@ the following list:
+ 1. Let |openInfo| be a [=new=] {{WebSocketOpenInfo}}.
+ 1. Set |openInfo| to «[ "{{WebSocketOpenInfo/extensions}}" → |extensions|,
+    "{{WebSocketOpenInfo/protocol}}" → |protocol|, "{{WebSocketOpenInfo/readable}}" → |readable|,
+    "{{WebSocketOpenInfo/writable}}" → |writable| ]».
+ 1. [=Resolve=] |stream|'s [=WebSocketStream/opened promise=] with |openInfo|.
+
+</div>
+
+<hr>
+
+<div algorithm="handle a WebSocketStream message">
+
+When [=a WebSocket message has been received=] for a {{WebSocketStream}} |stream| with type |type|
+and data |data|, the user agent must [=queue a task=] to follow these steps: [[!WSP]]
+
+ 1. If [=WebSocket/ready state=] is not {{WebSocket/OPEN}} (1), then return.
+ 1. Let |chunk| be determined by switching on |type|:
+
+  <dl class="switch">
+   : |type| indicates that the data is Text
+   :: a new {{DOMString}} containing |data|
+
+   : |type| indicates that the data is Binary
+   :: a new {{Uint8Array}} object, created in the [=relevant Realm=] of the {{WebSocketStream}}
+      object, whose contents are |data|
+  </dl>
+
+ 1. [=ReadableStream/Enqueue=] |chunk| into |stream|'s [=WebSocketStream/readable stream=].
+ 1. Apply backpressure to the WebSocket.
+
+ Note: Applying backpressure <span class=allow-2119>should</span> result in no new WebSocket
+ messages being handled until the backpressure is released. After an implementation-defined amount
+ of data is buffered, the client will stop reading from the underlying network connection resulting
+ in network-layer backpressure being applied.
+</div>
+
+<hr>
+
+When [=the WebSocket closing handshake is started=] for a {{WebSocketStream}} |stream|, the user
+agent must [=queue a task=] to change the |stream|'s [=WebSocketStream/ready state=] to
+{{WebSocket/CLOSING}} if it is not already. [[!WSP]]
+
+<hr>
+
+<div algorithm="handle WebSocketStream connection close">
+
+When [=the WebSocket connection is closed=] for a {{WebSocketStream}} |stream|, possibly
+ [=cleanly=], the user agent must [=queue a task=] to run the following substeps:
+
+ 1. Change the [=WebSocketStream/ready state=] to {{WebSocket/CLOSED}} (3).
+ 1. If |stream|'s [=WebSocketStream/handshake aborted flag=] is set, then return.
+ 1. If |stream|'s [=WebSocketStream/was ever connected flag=] is not set, then [=reject=] |stream|'s
+    [=opened promise=] with a [=new=] WebSocketError.
+ 1. Let |code| be [=the WebSocket connection close code=].
+ 1. Let |reason| be the result of applying [=UTF-8 decode without BOM=] to [=the WebSocket
+    connection close reason=].
+ 1. If the connection was closed [=cleanly=],
+  1. [=ReadableStream/Close=] |stream|'s [=WebSocketStream/readable stream=].
+  1. [=WritableStream/Error=] |stream|'s [=WebSocketStream/writable stream=] with an
+     "{{InvalidStateError}}" {{DOMException}} indicating that a closed {{WebSocketStream}} cannot be
+     written to.
+  1. [=Resolve=] |stream|'s [=closed promise=] with a [=new=] {{WebSocketCloseInfo}} constructed
```

Comment:
Same issue as above with dictionary [=new=] usage and map literal notation.

Reply:
Fixed, thanks.